core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/datarepo/HttpDataRepoDAO.scala

```diff
@@ -25,6 +28,11 @@ class HttpDataRepoDAO(dataRepoInstanceName: String, dataRepoInstanceBasePath: St
 
   override def getSnapshot(snapshotId: UUID, accessToken: OAuth2BearerToken): SnapshotModel = {
     // future enhancement: allow callers to specify the list of SnapshotRetrieveIncludeModel to retrieve
-    getRepositoryApi(accessToken).retrieveSnapshot(snapshotId, java.util.Collections.emptyList())
+    addDataRepoRowId(getRepositoryApi(accessToken).retrieveSnapshot(snapshotId, java.util.Collections.emptyList()))
+  }
+
+  //Snapshots-by-reference always have a datarepo_row_id, but that is not included in the model and should be
+  private def addDataRepoRowId(snapshot: SnapshotModel): SnapshotModel = {
+    snapshot.tables(snapshot.getTables().stream().map(t => t.addColumnsItem(datareporow_id)).collect(Collectors.toList()))
```

Comment:
`snapshot.getTables()` is specifically a Java list, so I had to do some Java manipulation here instead of pure Scala